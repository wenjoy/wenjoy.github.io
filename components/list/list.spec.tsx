import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { List } from '.';

describe('List', () => {
  it('should renders empty result', () => {
    const data: any[] = []
    const { container } = render(<List data={data} />)
    expect(container).toMatchSnapshot()

    const noResult = screen.getByText('Nothing can be displayed')

    expect(noResult).toBeInTheDocument()
  })

  it('should renders List correctly', () => {
    const data: any[] = [{ name: 'test one' }]
    const { container } = render(<List data={data} />)
    const result = screen.getByText('test one')
    expect(result).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})