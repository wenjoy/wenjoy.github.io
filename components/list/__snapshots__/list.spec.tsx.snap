// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should renders List correctly 1`] = `
<div>
  <div>
    <div>
      test one
    </div>
  </div>
</div>
`;

exports[`List should renders empty result 1`] = `
<div>
  <div
    data-testid="no-result"
  >
    Nothing can be displayed
  </div>
</div>
`;
